### DISCUSSION TITLE
Why create project in the BBN get warning error "Project count not found for project ‘webapps/testproject" in the backend log?

### DISCUSSION THREAD

**Discussion by QXgu** [BlueBrain/nexus#2777] [Q&A] [answered]
Hi,

I have a question about creating the project in the BBN fusion. We got the warning message shown below after creating the project but the project can be created successfully. Why we got that kind of warning message? Will that cause any problems for using the project? We met the problem that we can not find any data on the Search page while the query editor works fine. Is that possible that creating a project problem leads to the search resources not being found?

The version we used for BBN is 1.5.1

`2021-09-10 19:31:32 WARN c.e.b.n.d.sdk.ProgressesStatistics - Project count not found for project ‘webapps/testproject’
2021-09-10 19:31:32 INFO a.p.cassandra.query.EventsByTagStage - [6b2c6c59-3799-42b9-a3a4-28e953e28134]: EventsByTag query [project=webapps/testproject] starting with EC delay 4000ms: fromOffset [b165d960-126d-11ec-8080-808080808080 (2021-09-10 19:31:27:862)] toOffset [None]
2021-09-10 19:31:32 INFO a.p.cassandra.query.EventsByTagStage - [e1897a9e-2913-40b6-aaa2-9f8b9d059c67]: EventsByTag query [project=webapps/testproject] starting with EC delay 4000ms: fromOffset [b165d960-126d-11ec-8080-808080808080 (2021-09-10 19:31:27:862)] toOffset [None]`

Also, I found out that when I go to the default query page and hovered on the **project** tag, the page becomes blank. I met the same problem in the SandBox as well. Here is the error message from the frontend:
![image](https://user-images.githubusercontent.com/86449881/132916402-38adbbb3-4b0b-4a02-b6c3-27cf14abecb8.png)
![image](https://user-images.githubusercontent.com/86449881/132916616-bf61b517-4d49-4308-a8cc-3130e118ae9a.png)
![image](https://user-images.githubusercontent.com/86449881/132916044-7f2d0858-9009-4563-85fd-82ed6dc7a0c6.png)

We were guessing this might be the cause for the Search panel error we experienced. Please advise. Thanks

Thanks,
Amy

**Comment by umbreak** [comment]
About the error message on the backend side. Can you try the following:
```bash
# You have to set the env variables TOKEN and endpoint first.

curl -H "Authorization: Bearer $TOKEN" "{endpoint}/v1/projects/webapps/testproject/statistics"
```

This request is documented [here](https://bluebrainnexus.io/snapshot/docs/delta/api/projects-api.html#fetch-statistics)

If that returns some values different then 0, it works.

Alternatively, just reload the fusion page you mention and check on the logs if you still see the same error message. It shouldn't be there anymore.

What could have happened is that the first request to project counts was made before the cache we use to count projects was filled.

Please let us know if you have any further questions about the same issue.

**Reply by mnaghavi** [reply]
Hi @umbreak,
Thanks for the answer,

I've tried the command that you have suggested, and it answered other than 0.
So, it is working, as you mentioned, but the problem is that the error is still there with reloading the fusion page.

Another thing to mention here (maybe it helps for resolving this issue) is when you click on the project name (instead of hovering the mouse), the following error will appear:
```sh
{
  "@context": "https://bluebrain.github.io/nexus/contexts/error.json",
  "@type": "ResourceNotFound",
  "rejections": [
    {
      "@type": "ResourceNotFound",
      "reason": "Resource 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "Storage 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "File 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "ElasticSearch view 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "Blazegraph view 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "Composite view 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "Schema 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    },
    {
      "@type": "ResourceNotFound",
      "reason": "Resolver 'http://10.3.50.11/v1/projects/webapps/nexus-web' not found in project 'webapps/nexus-web'.",
      "status": 404
    }
  ]
}
```

And this is the screenshot of the nexus web interface after clicking on the project name:

<img width="1792" alt="Screen Shot 2021-09-16 at 5 01 41 PM" src="https://user-images.githubusercontent.com/13926694/133685120-bc836229-6025-4d21-9150-a9ebe7818299.png">

**Reply by mnaghavi** [reply]
JFTR, In the screenshot above, I was using `Delta=1.51` and `Fusion(web)=1.5.1`
Also, I have tested this in the latest available docker version for both, which is `Nexus Delta v1.6.0-M3+10-defb4f2c-SNAPSHOT` and `Nexus Fusion v1.6.0-M4`, but still the same issue happening with a different appearance:

<img width="1789" alt="Screen Shot 2021-09-17 at 1 21 20 PM" src="https://user-images.githubusercontent.com/13926694/133828801-c19900f4-f326-4d34-bf58-6d174331838e.png">

**Comment by dhaneshnm** [comment, marked as answer]
@mnaghavi 
So the "project" should not be listed here, because it is not a resource(like other things listed there). The UI error only means, that UI is trying to look it up as  a resource and is not able to find it (because it is not a resource).
This is a  bug in UI and will be fixed in next release. This display issue is not related to the original issue posted by @QXgu . For all practical purposes, you can ignore this behavior.
In future versions, project will not be listed inside "Project View" along with other resources, and you can see project details only in "project settings", which appears when you click on "Edit" button next to the project in project's listing page. (path : /admin/myOrgId). An example screenshot is given below:

![Screenshot 2021-09-20 at 11 03 50](https://user-images.githubusercontent.com/1479155/133978751-6352a3f3-b586-429b-8ab4-ffcec757ea57.png)

![Screenshot 2021-09-20 at 11 04 57](https://user-images.githubusercontent.com/1479155/133978823-4fe5cf49-4946-40bc-8cef-48efffe7326b.png)

**Reply by mnaghavi** [reply]
Thank you @dhaneshnm, for the explanation.
